selectel/mks.go

```diff
@@ -626,3 +628,119 @@ func parseMKSKubeVersionsV1Default(versions []*kubeversion.View) string {
 
 	return defaultVersion
 }
+
+// findQuota finds and returns quota for specified resource from quotas slice.
+func findQuota(quotas []*quotas.Quota, resource string) []quotas.ResourceQuotaEntity {
+	for _, q := range quotas {
+		if q.Name == resource {
+			return q.ResourceQuotasEntities
+		}
+	}
+
+	return nil
+}
+
+func checkQuotasForCluster(projectQuotas []*quotas.Quota, region string, zonal bool) error {
+	clusterQuotaChecked := false
+
+	var quota []quotas.ResourceQuotaEntity
+
+	if zonal {
+		quota = findQuota(projectQuotas, "mks_cluster_zonal")
+	} else {
+		quota = findQuota(projectQuotas, "mks_cluster_regional")
+	}
+
+	if quota == nil {
+		return errors.New("unable to find neither mks_cluster_zonal nor mks_cluster_zonal quotas")
+	}
+
+	for _, v := range quota {
+		if v.Region == region {
+			if v.Value-v.Used <= 0 {
+				if zonal {
+					return errors.New("not enough quota to create zonal k8s cluster")
+				}
+
+				return errors.New("not enough quota to create regional k8s cluster")
+			}
+			clusterQuotaChecked = true
+		}
+	}
+	if !clusterQuotaChecked {
+		return errors.New("unable to check neither regional nor zonal k8s cluster quota for a given region")
+	}
+
+	return nil
+}
+
+func checkQuotasForNodegroup(projectQuotas []*quotas.Quota, nodegroupOpts *nodegroup.CreateOpts) error {
+	cpuQuotaChecked, ramQuotaChecked, diskQuotaChecked := false, false, false
+
+	cpuQuota := findQuota(projectQuotas, "compute_cores")
+	if cpuQuota == nil {
+		return errors.New("unable to find CPU quota")
+	}
+	ramQuota := findQuota(projectQuotas, "compute_ram")
+	if ramQuota == nil {
+		return errors.New("unable to find RAM quota")
+	}
+
+	var volumeQuota []quotas.ResourceQuotaEntity
+	if nodegroupOpts.LocalVolume {
+		volumeQuota = findQuota(projectQuotas, "volume_gigabytes_local")
+	} else {
+		switch strings.Split(nodegroupOpts.VolumeType, ".")[0] {
```

Comment:
As I can see,  here we can get in trouble.
For instance: user  can define volume-type like a fast-ru-x, so in this case 0 element will be empty. I think that we should check that volume type is parseble.

Reply:
[According to docs](https://pkg.go.dev/strings#Split) resulting array always will have at least one element that contains given string. So in worst case here we will get nil in `volumeQuota` (because `findQuota` won't find any for an incorrect input string) and error will be thrown on 703 line